pandas/core/generic.py

```diff
@@ -5909,6 +5909,97 @@ def infer_objects(self: FrameOrSeries) -> FrameOrSeries:
             )
         ).__finalize__(self)
 
+    def as_nullable_dtypes(
+        self: FrameOrSeries, convert_integer: bool_t = True
+    ) -> FrameOrSeries:
+        """
+        Convert columns of DataFrame or a Series to types supporting ``pd.NA``.
+
+        If the dtype is ``object``, if possible, convert to ``StringDtype``,
+        ``BooleanDtype`` or an appropriate integer extension type, otherwise leave as
+        ``object``.
+
+        If the dtype is integer", convert to an appropriate integer type.
+
+        If the dtype is numeric, and consists of all integers, convert to an
+        appropriate integer extension type.
+
+        Parameters
+        ----------
+        convert_integer : bool, default True
+            Whether ``int`` types should be converted to integer extension types
+
+        Returns
+        -------
+        converted : same type as input object
```

Comment:
Can you run `python scripts/validate_docstrings.py DataFrame.as_nullable_dtypes` on this? I think this is supposed to be the type returned (Series or Frame, so may need to use Appender) and the description goes on the next line.

Reply:
OK, I fixed that.  Was following the pattern for `infer_objects` which doesn't pass that test!